src/renderers/shared/stack/reconciler/ReactCompositeComponent.js

```diff
@@ -833,11 +835,27 @@ var ReactCompositeComponentMixin = {
       this._currentElement = nextParentElement;
       this._context = nextUnmaskedContext;
       inst.props = nextProps;
-      inst.state = nextState;
+      this._updateState(nextState);
       inst.context = nextContext;
     }
   },
 
+  _updateState(nextState) {
+    this._instance.state = nextState;
+    this._notifyStateChanged();
+  },
+
+  // Explicitly notify about state change everytime the state changes,
+  // and don't count on _processPendingState, in case it might be called
+  // without actually changing the state later
+  _notifyStateChanged() {
```

Comment:
I still think we don't want this function. It will be empty in production. Let's just inline those calls.